homeassistant/components/jellyfin/media_player.py

```diff
@@ -0,0 +1,489 @@
+    ) -> BrowseMedia:
+        """Return a BrowseMedia instance.
+
+        The BrowseMedia instance will be used by the "media_player/browse_media" websocket command.
+
+        """
+
+        def browse_media_from_item(media: Any) -> BrowseMedia:
+            content_type_map = {
+                "Series": MEDIA_TYPE_TVSHOW,
+                "Movie": MEDIA_TYPE_MOVIE,
+                "CollectionFolder": "collection",
+                "Folder": "library",
+                "BoxSet": "boxset",
+            }
+            media_class_map = {
+                "Series": MEDIA_CLASS_DIRECTORY,
+                "Movie": MEDIA_CLASS_MOVIE,
+                "CollectionFolder": MEDIA_CLASS_DIRECTORY,
+                "Folder": MEDIA_CLASS_DIRECTORY,
+                "BoxSet": MEDIA_CLASS_DIRECTORY,
+                "Episode": MEDIA_CLASS_EPISODE,
+                "Seaoson": MEDIA_CLASS_SEASON,
+            }
+            return BrowseMedia(
+                title=media["Name"],
+                media_content_id=media["Id"],
+                media_content_type=content_type_map.get(media["Type"], MEDIA_TYPE_NONE),
+                media_class=media_class_map.get(media["Type"], MEDIA_CLASS_DIRECTORY),
+                can_play=True,
+                can_expand=media["Type"] not in ["Movie", "Episode"],
+                children_media_class="",
+                thumbnail=str(
+                    self._client.jellyfin.artwork(media["Id"], "Primary", 500)
+                ),
+                children=[],
+            )
+
+        async def create_item_children(
+            hass: HomeAssistant, client: JellyfinClient, user: str, itemid: str
+        ) -> list[BrowseMedia]:
+            children = await hass.async_add_executor_job(
+                lambda: dict(
+                    client.jellyfin.items(params={"parentId": itemid, "userId": user})
+                )
+            )
+
+            return [browse_media_from_item(item) for item in children["Items"]]
+
+        async def create_item_response(
+            hass: HomeAssistant, client: JellyfinClient, user: str, itemid: str
+        ) -> BrowseMedia:
+            items = await hass.async_add_executor_job(
+                lambda: dict(
+                    client.jellyfin.items(params={"ids": [itemid], "userId": user})
+                )
+            )
+
+            if not items or "Items" not in items or len(items["Items"]) < 1:
+                raise NotFoundErr()
+
+            return browse_media_from_item(items["Items"][0])
+
+        async def create_root_response(
+            hass: HomeAssistant, client: JellyfinClient, user: str
+        ) -> BrowseMedia:
+            folders = await hass.async_add_executor_job(
+                client.jellyfin.get_media_folders
+            )
+
+            children = [
+                await create_item_response(hass, client, user, folder["Id"])
+                for folder in folders["Items"]
```

Comment:
You may want to add a check for supported folder types here. At least photo libraries do not seem to work at the moment. You can browse, but when selecting a photo a 503 error is thrown (in my case at least).

Reply:
This filters out the image and music libraries now.

I might add support for those later, if I set them up on my instance.
Though I doubt that the image server is better than photoprism :)